BEP18.md

```diff
@@ -0,0 +1,123 @@
+
+| Field          | Type        | Description                                                                                                                                                                                                                                                                                                                                                      |
+|----------------|-------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Version        | int32       | snapshot version                                                                                                                                                                                                                                                                                                                                                 |
+| Height         | int64       | height of this snapshot                                                                                                                                                                                                                                                                                                                                          |
+| StateHashes    | []SHA256Sum | hashes of tendermint state chunks                                                                                                                                                                                                                                                                                                                                |
+| AppStateHashes | []SHA256Sum | hashes of app state chunks                                                                                                                                                                                                                                                                                                                                       |
+| BlockHashes    | []SHA256Sum | hashes of the block in this snapshot, currently only one block is synced                                                                                                                                                                                                                                                                                             |
+| NumKeys        | []int64     | number of keys for each sub-store, this sacrifices clear boundary between cosmos and tendermint, saying tendermint knows application DB might be organized by sub-stores.<br><br> |
```

Comment:
What if the order or number of sub-stores changes? Or this is controlled by Version (snapshot version)?

Reply:
> What if the order or number of sub-stores changes? Or this is controlled by Version (snapshot version)?

State sync will accept majority peer's manifest. Saying if the order or length of `NumKeys` of manifest in some nodes is inconsistent with the majority node (maybe they forget to upgrade their binary whereas others changed the order or num of sub stores), the new joined node won't sync from it (accept his manifest file).